src/ethereum_test_tools/rpc/rpc.py

```diff
@@ -92,6 +92,12 @@ def get_transaction_count(self, address: Address, block_number: BlockNumberType
         block = hex(block_number) if isinstance(block_number, int) else block_number
         return self.post_request("eth_getTransactionCount", [address, block])
 
+    def get_transaction_by_hash(self, transaction_hash: str):
```

Comment:
It would be amazing if we could get he return type to match the `Transaction` pydantic type. Can also do this in the future, when refining the eth rpc classes!